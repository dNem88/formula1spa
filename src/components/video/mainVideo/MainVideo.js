import React, {Fragment} from 'react'
import styles from './mainVideo.module.css'
import Video from '../../common/video/Video'
import VideoInfo from '../../common/videoInfo/VideoInfo'
import {useOutletContext} from 'react-router-dom'
import Fieldset from '../../common/containers/fieldset/Fieldset'

function MainVideo() {
    const context = useOutletContext()
    let video = null;
    if ( context.mainVideo) {
        video = context.mainVideo
        console.log(video)
    }
   
    return (
        <Fieldset content={'MUST SEE'} size={'half'} color={'white'}>
            {video ?
             <Fragment>
                <div className={styles['video-wrapper']}>
                    <Video url={video.url}/>
                    <VideoInfo _id={video._id} title={video.title}/>
                </div>
             </Fragment> : null
            } 
        </Fieldset>
    )
}

export default MainVideo;