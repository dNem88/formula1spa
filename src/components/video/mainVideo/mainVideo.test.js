import React from 'react'
import {MemoryRouter, Outlet, useOutletContext} from 'react-router-dom'
import renderer from 'react-test-renderer'
import MainVideo from './MainVideo'



jest.mock('../../common/video/Video', () => () => 'Video')
jest.mock('../../common/videoinfo/VideoInfo', () => () => 'VideoInfo')


describe('Testing MainVideo', () => {
    
    it('Render Properly', () => {
        const snapshot = renderer.create(
        <MemoryRouter>
            <Outlet context={{mainVideo: 'mock'}}>
                <MainVideo context={{mainVideo: 'mock'}}/>
            </Outlet>
        </MemoryRouter>
    ).toJSON()
        expect(snapshot).toMatchSnapshot()
        // expect(snapshot.type).toBe('section')
        // expect(snapshot.props.className).toBe('main-container')
    })
})