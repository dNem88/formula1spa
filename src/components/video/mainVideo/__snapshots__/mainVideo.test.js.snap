// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MainVideo Render Properly 1`] = `null`;
